src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,35 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction(SProcService.WriteTransaction serviceWriteTransaction) {
+                if (serviceWriteTransaction == null) {
```

Comment:
I really do not understand why you are trying to put this logic in here. It somehow does not belong to the enum code for me.

And I do not understand why you cannot check that `WriteTransaction` is `USE_FROM_SERVICE` down in the code, where the decision is being done. Maybe I am missing something here.